src/main/java/AdventOfCode/DayFourAdvent.java

```diff
@@ -0,0 +1,143 @@
+            String[] keys = file.get(i).split("[,: ]");
+            for (int j = 0; j < keys.length; j += 2) {
+                if (keys[j].equals("eyr")) {
+                    eyr = keys[j + 1];
+                    if ((Integer.parseInt(eyr) >= 2020) && (Integer.parseInt(eyr) <= 2030)) {
+                        eyr = keys[j + 1];
+                    } else {eyr = null;}
+                }
+                if (keys[j].equals("iyr")) {
+                    iyr = keys[j + 1];
+                    if ((Integer.parseInt(iyr) >= 2010) && (Integer.parseInt(iyr) <= 2020)) {
+                        iyr = keys[j + 1];
+                    } else { iyr = null;}
+                }
+                if (keys[j].equals("hcl")) {
+                    if (keys[j+1].matches("^#[0-9a-f]{6}") && keys[j+1].length() == 7) {
+                            hcl = keys[j + 1];
+                    } else {hcl = null;}
+                }
+                if (keys[j].equals("byr")) {
+                    byr = keys[j + 1];
+                    if ((Integer.parseInt(byr) >= 1920) && (Integer.parseInt(byr) <= 2002)) {
+                        byr = keys[j + 1];
+                    } else {byr = null;}
+                }
+                if (keys[j].equals("ecl")) {
+                    ecl = keys[j + 1];
+                    if (ecl.equals("amb") || ecl.equals("blu") || ecl.equals("brn") || ecl.equals("gry") || ecl.equals("grn") || ecl.equals("hzl")
+                    || ecl.equals("oth")) {
+                        ecl = keys[j + 1];
+                    } else {
+                        ecl = null;
+                    }
+                }
+                if (keys[j].equals("hgt")) {
+                    hgt = keys[j + 1];
+                    String [] numberInString = hgt.split("[cm in]");
+                    int number = Integer.parseInt(numberInString[0]);
+                    if ((hgt.endsWith("cm") && (number >= 150 && number <= 193)) ||
+                            (hgt.endsWith("in") && (number >= 59 && number <= 76))) {
+                        hgt = keys[j + 1];
+                    } else {
+                        hgt = null;
+                    }
+                }
+                if (keys[j].equals("pid")) {
+                    pid = keys[j + 1];
+                    if (pid.length() == 9) {
+                        pid = keys[j + 1];
+                    } else
+                    pid = null;
+                }
+            }
+            if (keys[0].equals("") || i == file.size() - 1) {
+                if (byr != null && iyr != null && eyr != null && hgt != null && pid != null && hcl != null && ecl != null) {
+                    validPassports++;
+                }
+                byr = null;
+                iyr = null;
+                eyr = null;
+                hgt = null;
+                pid = null;
+                hcl = null;
+                ecl = null;
+            }
+        }
+        return validPassports;
```

Comment:
Split into functions. You can make at least 7 functions out of this, which would make it significantly more readable.